analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ObjectsShouldNotBeDisposedMoreThanOnceBase.cs

```diff
@@ -18,10 +18,31 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
 
 public abstract class ObjectsShouldNotBeDisposedMoreThanOnceBase : SymbolicRuleCheck
 {
     protected const string DiagnosticId = "S3966";
-    protected const string MessageFormat = "Refactor this code to make sure '{0}' is disposed only once.";
+    protected const string MessageFormat = "Resource '{0}' has already been disposed explicitly or implicitly through a using statement. Please remove the redundant disposal.";
+
+    private static readonly string[] DisposeMethods = { "Dispose", "DisposeAsync"};
+
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        var state = context.State;
+        if (context.Operation.Instance.AsInvocation() is { } invocation && DisposeMethods.Contains(invocation.TargetMethod.Name))
```

Comment:
When done for both C# and VB, then the `Equals()` method needs to be used with the `Language.NameComparison` parameter.
`DisposeMethods.Any(x => x.Equals(invocation.TargetMethod.Name, Language.NameComparison))`

Reply:
When I checked the MSDN the [Dispose ](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose) and [DisposeAsync ](https://learn.microsoft.com/en-us/dotnet/api/system.iasyncdisposable.disposeasync?view=net-7.0) were the same case in both C# and VB.NET.
So IMO it's not needed to overengineer this in this case.

Reply:
VB.NET is case insensitive. `x.Dispose()` is just as valid as `x.DISPOSE()`
Using `Language.NameComparison` makes sure that case-insensitive name comparison will be used in VB, but case sensitive in C#.
It's worth adding a test case for this scenario.

Reply:
I did not know that! Thanks!

Reply:
It's testing ISymbol name, so case sensitive comparison is not needed.
The problem here (for VB phase) is, that method implementing Dispose method can have a different name
```
Public Sub Close() Implements IDisposable.Dispose
End Sub
```
or even be private (that's not a problem ,because only cast can call it)
```
Private Sub IDisposable_Dispose() Implements IDisposable.Dispose
End Sub
```